test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/test/txsim"
+	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/stretchr/testify/require"
+)
+
+// This will only run tests within the v1 major release cycle
+const MajorVersion = 1
+
+func TestMinorVersionCompatibility(t *testing.T) {
+	if os.Getenv("E2E") == "" {
+		t.Skip("skipping e2e test")
+	}
+
+	if os.Getenv("E2E_VERSION") == "" {
+		t.Skip("skipping e2e test: E2E_VERSION not set")
+	}
+
+	versionStr := os.Getenv("E2E_VERSION")
+	versions := ParseVersions(versionStr).FilterMajor(MajorVersion).FilterOutReleaseCandidates()
+	numNodes := 4
+	r := rand.New(rand.NewSource(seed))
+	t.Log("Running minor version compatibility test", "versions", versions)
```

Comment:
[question] does this test also include patch versions?

^ disregard. Answered my own question by modifying a unit test below. This does include patch versions in the major version.